pkg/storage/storage.go

```diff
@@ -194,18 +194,22 @@ type TreeEntry struct {
 }
 
 //Tree returns the files and folders at a given rev at a path in a repository
-func (s *storage) Tree(ctx context.Context, owner string, name string, rev string, path string) ([]TreeEntry, error) {
+func (s *storage) Tree(ctx context.Context, owner, name, rev, path string) ([]TreeEntry, error) {
 	args := []string{"ls-tree", rev, path}
 	cmd := exec.CommandContext(ctx, s.git, args...)
 	cmd.Dir = filepath.Join(s.root, owner, name)
-	out, err := cmd.Output()
+	out, err := cmd.StdoutPipe()
 	if err != nil {
-		println(string(out))
 		return nil, errors.Wrap(err, "failed to run git ls-tree")
 	}
+	if err = cmd.Start(); err != nil {
```

Comment:
for `cmd.Start()` you need to `cmd.Wait()` after the scanner-loop :slightly_smiling_face: